src/controller.py

```diff
@@ -133,6 +135,27 @@ def save_planes_on_close(self):
         sett().slicing.splanes_file = path.basename(splanes_full_pth)
         save_settings()
 
+    def choose_printer_path(self):
+        printer_path = QFileDialog.getExistingDirectory(
+            self.view,
+            locales.getLocale().ChoosePrinterDirectory,
+            sett().hardware.printer_dir
+        )
+
+        if printer_path:
+            sett().hardware.printer_dir = printer_path
+            # calibration file will be at default location
+            sett().hardware.calibration_file = "calibration_data.csv"
+            
+            # save settings
+            save_settings()
+
+            # update label with printer path
+            self.view.printer_path_edit.setText(os.path.basename(printer_path))
+
+            # update path in calibration model
+            self.calibrationController.updateCalibrationFilepath(PathBuilder.calibration_file())
```

Comment:
Maybe add
```python
if self.calibrationController:
``` 
but for this in case hardware module is not availabe we will need to set
```python
self.calibrationController = None
``` 
or the check may be
```python
if gettattr(self, calibrationController, None):
```